src/H5private.h

```diff
@@ -1561,6 +1561,15 @@ H5_DLL int HDvasprintf(char **bufp, const char *fmt, va_list _ap);
 #define HDwrite(F, M, Z) write(F, M, Z)
 #endif
 
+/*
+ * Ugly hack to cast away const for freeing const-qualified pointers.
+ * Should only be used sparingly, where the alternative (like keeping
+ * an equivalent non-const pointer around) is far messier messier.
+ */
+#ifndef HDfree_const
+#define HDfree_const(mem) HDfree((void *)(uintptr_t)mem)
+#endif
```

Comment:
Hmm ok if that's only used in tests, I think you should leave it in tests because having that in H5private.h seems ugly

Reply:
I don't really like the way that this works, but we also do the same thing in H5MM_xfree_const. Right now it's only used in the tests so I only had it in the one test file, but moved it to H5private.h from Dana's suggestion. I can move it back if we don't really see it being useful elsewhere.

Reply:
Not being able to free const pointers is often considered to be a deficiency in the standard. If we leave it here, it shouldn't get HD as that implies it's a part of the platform-independence stuff in H5private.h. Maybe h5_free_const to go with the other tools functionality.

Reply:
Decided to move it to h5test.h for now, since the tests are where this macro will most likely be used. We can always move it back to H5private.h in the future if we use it more widely.